### PR TITLE
SolverVault liquidation and insolvency integration tests

### PR DESCRIPTION
**Pull request by prateekdefi**
Issue: https://linear.app/perennial/issue/PE-2149/[testing]-add-liquidation-and-insolvency-integration-tests-to

### CODE REVIEW

File: packages/vault/test/integration/vault/SolverVault.test.ts [hunk 1]
```diff
@@ -1584,6 +1584,439 @@ describe('SolverVault', () => {
...
+          // Ensure maintenance requirement is violated.
+          const collateral = (await btcMarket.locals(vault.address)).collateral
+          const maintenanceRatio = (await btcMarket.riskParameter()).maintenance
+          const maintenanceRequired = (await btcMarket.positions(vault.address)).short
+            .mul(newBtcOraclePrice)
+            .mul(maintenanceRatio)
+            .div(1e12)
+          expect(collateral).to.be.lessThan(maintenanceRequired)
+
+          // Redeem shares to close position
```
Comment (kbrizzle) [inline packages/vault/test/integration/vault/SolverVault.test.ts:1676]: I think we want to maintain the same flow / checks as the `MakerVault` equivalent tests here.

What was reasoning for switching this to testing redeeming a liquidatable position, instead of testing actually liquidating the position and the aftermath?

Comment (prateekdefi) [reply packages/vault/test/integration/vault/SolverVault.test.ts:1676]: Reasoning to add redeem was that I checked that liquidation will be successful by `expect(collateral).[to.be](http://to.be/).lessThan(maintenanceRequired)` and I thought of testing whether collateral from a liquidatable vault can be redeemed. Now that I think I could have added different tests for both of these two scenarios.

Comment (prateekdefi) [reply packages/vault/test/integration/vault/SolverVault.test.ts:1676]: I will update this current test to actually liquidating the position and also create a new test for redeeming the liquidatable position similar to `rebalance position to redeem liquidatable vault` in `MakerVault`.

Comment (kbrizzle) [reply packages/vault/test/integration/vault/SolverVault.test.ts:1676]: Ahh I see where you sourced these from now actually. We can actually skip the first two `short exposure` tests related to rebalancing (those are specific to the MakerVault strategy).

The two `recovers from...` tests in each group are the ones we should be referencing. Because we want to make sure if one of the positions gets liquidated, the Vault still operates all the edge cases as expected (with shortfall as well).

**Review by kbrizzle — DISMISSED** [review]

**Review by EdNoepel — APPROVED** [review]
LGTM; will also need to merge this to v2.5.
